vscode/eslint.config.mjs

```diff
@@ -1,28 +1,91 @@
 import typescriptEslint from "@typescript-eslint/eslint-plugin";
 import tsParser from "@typescript-eslint/parser";
+import tsEslint from "typescript-eslint";
+import prettierRecommended from "eslint-plugin-prettier/recommended";
+import prettier from "eslint-plugin-prettier";
+import unusedImports from "eslint-plugin-unused-imports";
+import react from "eslint-plugin-react";
+import globals from "globals";
+import eslint from "@eslint/js";
 
-export default [{
-    files: ["**/*.ts"],
-}, {
+export default [
+  eslint.configs.recommended,
+  ...tsEslint.configs.recommended,
+  react.configs.flat.recommended,
+  prettierRecommended,
+  {
+    files: [
+      "**/*.js",
+      "**/*.jsx",
+      "**/*.cjs",
+      "**/*.mjs",
+      "**/*.ts",
+      "**/*.tsx",
+    ],
+  },
+  {
     plugins: {
-        "@typescript-eslint": typescriptEslint,
+      "@typescript-eslint": typescriptEslint,
+      prettier,
+      "unused-imports": unusedImports,
+      react,
     },
 
     languageOptions: {
-        parser: tsParser,
-        ecmaVersion: 2022,
-        sourceType: "module",
+      parser: tsParser,
+      ecmaVersion: 2023, // keep in sync with tsconfig.json
+      sourceType: "module",
+      parserOptions: {
+        ecmaFeatures: {
+          jsx: true,
+        },
+      },
+      globals: {
+        ...globals.es2023,
+        ...globals.node,
+        myCustomGlobal: "readonly",
+      },
     },
 
     rules: {
-        "@typescript-eslint/naming-convention": ["warn", {
-            selector: "import",
-            format: ["camelCase", "PascalCase"],
-        }],
+      "@typescript-eslint/naming-convention": [
+        "warn",
+        {
+          selector: "import",
+          format: ["camelCase", "PascalCase"],
+        },
+      ],
 
-        curly: "warn",
-        eqeqeq: "warn",
-        "no-throw-literal": "warn",
-        semi: "warn",
+      curly: "warn",
+      eqeqeq: "warn",
+      "no-throw-literal": "warn",
+      semi: "warn",
+
+      "prettier/prettier": ["warn"],
+      "@typescript-eslint/no-unused-vars": [
+        "warn",
+        { argsIgnorePattern: "^_" },
+      ],
+      "unused-imports/no-unused-imports": ["warn"],
+      "@typescript-eslint/no-explicit-any": "warn",
+      "react/jsx-key": "warn",
+      // "react-hooks/rules-of-hooks": "warn",
+      // "react-hooks/exhaustive-deps": "warn",
+      "no-extra-boolean-cast": "warn",
+      "prefer-const": "warn",
+      "react/no-unknown-property": ["warn"],
```

Comment:
Should adjust most of these back to error since they're warn more because the state of tackle2-ui's code.  Probably not as many issues in the current code base here. ;-)

Reply:
VS Code API is loosely typed and rules like `no-explicite-any` would cause more harm then benefits if set to error level - check i.e. [Command interface](https://github.com/microsoft/vscode/blob/0f9abf3c3567e53df749299dafcc988831a3b4e7/src/vscode-dts/vscode.d.ts#L40)